src/core/lib/security/credentials/s2a/grpc_s2a_credentials_options.h

```diff
@@ -0,0 +1,93 @@
+/*
+ *
+ * Copyright 2019 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#ifndef GRPC_CORE_LIB_SECURITY_CREDENTIALS_S2A_GRPC_S2A_CREDENTIALS_OPTIONS_H
+#define GRPC_CORE_LIB_SECURITY_CREDENTIALS_S2A_GRPC_S2A_CREDENTIALS_OPTIONS_H
+
+#include <grpc/grpc_security.h>
+#include <grpc/support/port_platform.h>
+
+namespace experimental {
+
+enum s2a_supported_ciphersuite {
+  AES_128_GCM_SHA256 = 0,
+  AES_256_GCM_SHA384 = 1,
+  CHACHA20_POLY1305_SHA256 = 2,
+};
+
+/** An ordered list of ciphersuites for use by the S2A. **/
+typedef struct s2a_ciphersuite {
+  struct s2a_ciphersuite* next;
+  s2a_supported_ciphersuite cipher;
```

Comment:
It seems a little overkill for using linked list here since we will only have three values at most. Can we use an array (of three elements) and keep track of its actual size?

Reply:
Good point. I replaced this with a `std::vector` as a member variable of the `grpc_s2a_credentials_options` class. (Note that we are allowed to use `std::vector` now in the C core.)